app/helpers/visualizations_helper.rb

```diff
@@ -51,4 +58,116 @@ def get_form_filter_types
         ]
     end
 
+    def get_form_summarize_methods
+        [
+            ["count"],
+            ["sum"],
+            ["maximum"],
+            ["minimum"]
+        ]
+    end
+
+    def get_chart_info(visualization_id)
+        data = Student.all;
+        visualization = Visualization.find(visualization_id)
+        chart_type = visualization.chart_type
+
+        # Repeatedly apply each filter on the dataset
+        visualization.filters.each_with_index do |filter, index|
+            data = apply_filter(data, filter)
+            puts ">> Applied filter #{index + 1}: #{filter.inspect}"
+        end
+
+        visualization.variables.each_with_index do |variable, index|
+            data = set_variable(data, variable, chart_type)
+            puts ">> Set variable #{index + 1} : #{variable.inspect}"
+        end
+
+        chart_info = Hash.new
+        chart_info[:data] = data
+        chart_info[:chart_type] = chart_type
+        chart_info[:options] = get_chart_options(visualization_id)
+        # Render visualization partial. This partial should have logic to determine which chart to graph
+        return chart_info
+    end
+
+
+
+    # Applies a single filter on the dataset
+    def apply_filter(data, filter_model)
+        filter_type = filter_model.filter_type
+        variable_name = filter_model.variable_name;
+        value1 = filter_model.value1
+        value2 = filter_model.value2
+
+        case filter_type
+            when 'equals_to'
+                filter_hash = Hash.new
+                filter_hash[variable_name] = value1
+                data = data.where(filter_hash)
+
+            when 'greater_than'
+                data = data.where("#{variable_name} > ?", variable_name, value1)
```

Comment:
Why does data.where take variable_name as an argument when it doesn't in the following code?

Reply:
Yuhp, that's a mistake! Thanks for catching that👍🏽

Reply:
We'll fix this and test things out when we call. Hopefully, we can get this done on your end.